request-manager/auth/auth.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2018, Square, Inc.
+
+// Package auth provides request authentication and authorization. By default,
+// there is no auth; all callers and requests are allowed. The Plugin interface
+// allows user-defined auth in combination with user-defined request sepc ACLs.
+// See docs/auth.md.
+package auth
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/square/spincycle/proto"
+)
+
+// Caller represents an HTTP client making a request. Callers are determined by
+// the Plugin Authenticate method. The default Plugin (AllowAll) returns a zero
+// value Caller.
+type Caller struct {
+	// Name of the caller, whether human (username) or machine (app name). The
+	// name is user-defined and only used by Spin Cycle for logging and setting
+	// proto.Request.User.
```

Comment:
If we're keeping the SetUsername hook for now, the caller Name isn't being used for proto.Request.User.